cpp/benchmarks/copying/contiguous_split_benchmark.cu

```diff
@@ -70,18 +77,22 @@ void BM_contiguous_split(benchmark::State& state)
   int64_t num_rows        = total_desired_bytes / (num_cols * el_size);
 
   // generate input table
-  srand(31337);
-  auto valids = cudf::detail::make_counting_transform_iterator(0, [](auto i) { return true; });
+  auto valids = cudf::detail::make_counting_transform_iterator(
+    0, [] __device__(cudf::size_type i) { return true; });
   std::vector<cudf::test::fixed_width_column_wrapper<int>> src_cols(num_cols);
   for (int idx = 0; idx < num_cols; idx++) {
-    auto rand_elements =
-      cudf::detail::make_counting_transform_iterator(0, [](int i) { return rand(); });
+    auto rand_elements = cudf::detail::make_counting_transform_iterator(0, [idx] __device__(int i) {
+      thrust::default_random_engine rng(31337 + idx);
```

Comment:
I think curand is a bigger hammer than we really need here.

Reply:
Even without curand, we wouldn't want to compute and discard 100M elements on the last thread of a large table. I'd like to be sure that isn't happening here.

I also considered suggesting the [approach of thrust's Monte Carlo example](https://github.com/NVIDIA/thrust/blob/295025a533c64855483fbb928130fbcd3e61e1a5/examples/monte_carlo.cu#L12-L38) to generate a seed on each thread (rather than using a fixed seed and discarding `row_index` items), which would also be viable if we can combine the column index and row index in the seed for each thread.

Reply:
`discard` is a constant time operation on this generator: https://github.com/NVIDIA/thrust/blob/d35f44f36fc4730605d39abdfac92f500ff44a55/thrust/random/detail/linear_congruential_engine_discard.h#L49-L79

Even if it wasn't, the runtime is already only 10s, so I think any further work here is premature optimization.

Reply:
@jrhemstad Good find on the constant time discard. Thanks for looking into that.

I hear your concern about premature optimization. I expect we will re-use these patterns in subsequent PRs fixing other benchmarks so I wanted to make sure we are building with the right tools. We can resolve this since the discard is constant time. 👍